### ISSUE TITLE
Invalid imports not reported when importing from specific package (VSCode)

### ISSUE THREAD

**Issue by schperplata** [microsoft/pylance-release#300]
Issue Type: <b>Bug</b>

TODO: change the name of the issue to better represent the real issue.

Seems like Pylance reports warning for imports that are not found on the search path (sys.path?), but not if imports specify package and sub-package name. 

**1.** OK:
![image](https://user-images.githubusercontent.com/8736111/91665899-27942d00-eaf9-11ea-83fb-270f7418dc68.png)
Note: see that two warnings are reported, not one.

**2.** No warning issued (_unitest.tests_ does not exist, while _unittest.test_ does):
![image](https://user-images.githubusercontent.com/8736111/91665921-45619200-eaf9-11ea-90e7-c7ac686e067e.png)

**3.** No error/warning squigels appear, although we can see pylance discovered that `suites` does not exist - on mouse hover.
![image](https://user-images.githubusercontent.com/8736111/91665978-c0c34380-eaf9-11ea-8de7-6c2f75b68710.png)

Note: unittest module was selected only as an exmple. I've discovered this in my own project, where some import typo error was not reported. Root folder where `isys.generic` should be found was added via `PYTHONPATH`.
![image](https://user-images.githubusercontent.com/8736111/91666047-37604100-eafa-11ea-924e-e1b73b76f0ea.png)

Default settings are used.

Extension version: 2020.8.2
VS Code version: Code 1.47.3 (91899dcef7b8110878ea59626991a18c8a6a1b3e, 2020-07-23T13:12:49.994Z)
OS version: Windows_NT x64 10.0.18363

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-6820HQ CPU @ 2.70GHz (8 x 2712)|
|GPU Status|2d_canvas: enabled<br>flash_3d: enabled<br>flash_stage3d: enabled<br>flash_stage3d_baseline: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>oop_rasterization: disabled_off<br>protected_video_decode: unavailable_off<br>rasterization: enabled<br>skia_renderer: disabled_off_ok<br>video_decode: enabled<br>viz_display_compositor: enabled_on<br>viz_hit_test_surface_layer: disabled_off_ok<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|undefined|
|Memory (System)|15.92GB (8.39GB free)|
|Process Argv||
|Screen Reader|no|
|VM|0%|
</details>

**Comment by erictraut** [comment]
Thanks for filing the bug report.

The behaviors you're seeing are all intended. That doesn't mean that they are correct, but there was thinking that went into each of them.

First, some background. Pylance is used by a spectrum of Python users. Some simply want good completion suggestions to assist in coding. They tend to be annoyed by any false-positive errors, so Pylance disables most diagnostic rules by default. Those who are interested in more error reporting (sometimes at the expense of false positives) can set "python.analysis.typeCheckingMode" to "basic". In this mode, more diagnostic rules are enabled. Those who are interested in the strictest form of type checking can set the type checking mode to "strict", in which case all diagnostic rules are enabled. It looks like you have the type checking mode set to "off", so most of your diagnostic rules are disabled.

Now, for the specific issues you reported...

`import unittest.tests` refers to a submodule of `unittest`. Some packages use tricks at runtime to populate their package namespace. These tricks cannot be detected by a static analyzer, so there are cases where reporting an error in this situation would be a false positive. The current behavior suppresses the diagnostic if the import can be partially resolved. In your example, "unittest" was resolved, but "tests" was not. It is easy to disable this behavior and report diagnostics for cases where the import cannot be _fully_ resolved. I think it's worth making this change to see what feedback we receive. If false positives are sufficiently rare, then it's probably worth it to eliminate the false negatives.

@jakebailey, @huguesv — I made this change in the pyright repo. Let me know if you have any concerns here.

`from unittest.test import suites` is another case where false positives can and do occur because of the dynamic nature of some packages. If you enable the basic type checking mode, an error will be reported in this case.
